lib/Tumblr/StreamBuilder/StreamFilters/StreamFilter.php

```diff
@@ -133,11 +134,11 @@ final public function filter(
     /**
      * The filter concrete logic implementation.
      * @param StreamElement[] $elements The candidate StreamElement instances to filter.
-     * @param StreamFilterState $state The filter state passed from previous filter operations.
+     * @param StreamFilterState|null $state The filter state passed from previous filter operations.
      * @param StreamTracer|null $tracer The tracer passed in to track filter behaviors.
      * @return StreamFilterResult
      */
-    abstract protected function filter_inner(array $elements, StreamFilterState $state = null, StreamTracer $tracer = null): StreamFilterResult;
+    abstract protected function filter_inner(array $elements, ?StreamFilterState $state = null, ?StreamTracer $tracer = null): StreamFilterResult;
```

Comment:
why is this change needed ?

Reply:
@nicola-barbieri this change is unrelated to the goal of this PR. but the php docs were not consistent with the function arguments and I was seeing a warning on my local env. I fixed it since I was already modifying this class.